### PR TITLE
Test gap filling

### PR DESCRIPTION
**Pull request by YingXue**
filled up some gaps in tests
removed snapshot on mounted component, only test snapshot on the component itself to remove a lot of unecessary details in the snapshot
test coverge is now 85.08%

### CODE REVIEW

File: src/app/devices/deviceContent/components/deviceContent.spec.tsx [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+            ...overrides
+        };
+
+        return (
+            <DeviceContentComponent {...props} />
+        );
+    };
+
+    it('matches snapshot', () => {
+        testSnapshot(getComponent());
```
Comment (montgomp) [inline src/app/devices/deviceContent/components/deviceContent.spec.tsx]: would be good to add snapshot test after state changed for appMenuVisible

File: src/app/devices/deviceContent/components/deviceProperties/deviceProperties.spec.tsx [hunk 2]
```diff
@@ -59,22 +61,36 @@ describe('components/devices/deviceProperties', () => {
...
-        expect(wrapper).toMatchSnapshot();
+    it('shows Shimmer while loading', () => {
+        const wrapper = mountWithLocalization(
+            getComponent(), false, true, [pathname]
+        );
         expect(wrapper.find(Shimmer)).toBeDefined();
     });
 
+    it('matches snapshot while interface cannot be found', () => {
+        testSnapshot(getComponent({isLoading: false, twinWithSchema: undefined}));
```
Comment (montgomp) [inline src/app/devices/deviceContent/components/deviceProperties/deviceProperties.spec.tsx]: When I run the tests locally, I'm seeing not covered branch for when it should be returning `<InterfaceNotFoundMessageBoxContainer/>` so I don't think this is doing overriding the props as you'd expect.

File: src/app/devices/deviceContent/components/deviceSettings/deviceSetttingsPerInterfacePerSetting.spec.tsx [hunk 3]
```diff
@@ -54,7 +54,7 @@ describe('components/devices/deviceSettingsPerInterfacePerSetting', () => {
         settingModelDefinition: propertyModelDefinition,
         settingSchema: propertySchema};
 
-    it('matches snapshot when there is a writable property of simple type without sync status', () => {
+    it('renders when there is a writable property of simple type without sync status', () => {
```
Comment (montgomp) [inline src/app/devices/deviceContent/components/deviceSettings/deviceSetttingsPerInterfacePerSetting.spec.tsx]: Will need more tests to cover displayName and description returning '--' and unit returning something other than '--'.

File: src/app/devices/deviceContent/components/shared/complexReportedFormPanel.spec.tsx [hunk 4]
```diff
@@ -40,13 +40,10 @@ describe('complexReportedFormPanel', () => {
...
-        return mountWithLocalization(
-            <ComplexReportedFormPanel {...props} />
-        );
+        return (<ComplexReportedFormPanel {...props} />);
     };
 
     it('matches snapshot without twinWithSchema', () => {
-        const wrapper = getComponent();
-        expect(wrapper).toMatchSnapshot();
+        testSnapshot(getComponent());
```
Comment (montgomp) [inline src/app/devices/deviceContent/components/shared/complexReportedFormPanel.spec.tsx]: Would be good to add test where displayName is rendered as `'--'`

File: src/app/devices/deviceList/components/deviceList.spec.tsx [hunk 5]
```diff
@@ -5,48 +5,81 @@
...
-        <DeviceListComponent
-            connectionString="connectionString"
-            devices={devices}
-            listDevices={jest.fn()}
-            deleteDevices={jest.fn()}
-            isFetching={false}
-        />);
-
-        expect(wrapper).toMatchSnapshot();
+        testSnapshot(getComponent());
```
Comment (montgomp) [inline src/app/devices/deviceList/components/deviceList.spec.tsx]: Would be good to test snapshots with `query.continuationTokens` and `query.currentPageIndex`

Comment (montgomp) [reply src/app/devices/deviceList/components/deviceList.spec.tsx]: Also with rendering `lastActivityTime` and `statusUpdatedTime` rendering as `'--'`

File: src/app/devices/deviceList/components/deviceListCell.spec.tsx [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+    const getComponent = (overrides = {}) => {
+        const props = {
+            ...deviceListCellProps,
+            ...overrides,
+        };
+        return <DeviceListCell {...props} />;
+    };
+
+    it('matches snapshot while loading', () => {
+        testSnapshot(getComponent());
```
Comment (montgomp) [inline src/app/devices/deviceList/components/deviceListCell.spec.tsx]: Would be good to render additional snapshots with the various `context.t(ResourceKeys.deviceLists.noData)` branches being hit.

File: src/app/settings/components/repositoryLocationListItem.tsx [hunk 7]
```diff
@@ -63,7 +64,7 @@ export default class RepositoryLocationListItem extends React.Component<Reposito
                         </div>
                         {item.repositoryLocationType !== REPOSITORY_LOCATION_TYPE.Public && <IconButton
                             className="remove-button"
-                            iconProps={{ iconName: 'cancel' }}
+                            iconProps={{ iconName: CANCEL }}
```
Comment (montgomp) [inline src/app/settings/components/repositoryLocationListItem.tsx]: I have a couple more tests on my machine we can add to the spec file for this component.

File: src/app/shared/utils/testHelpers.tsx [hunk 8]
```diff
@@ -5,7 +5,9 @@
 import * as React from 'react';
 import { shallow, mount } from 'enzyme';
 import { MemoryRouter } from 'react-router-dom';
+import { Provider } from 'react-redux';
```
Comment (montgomp) [inline src/app/shared/utils/testHelpers.tsx]: we should ignore this file from test coverage and remove the `testWithLocalizationContextAndErrorBoundary` as it is not being called by any tests.

**Comment by YingXue** [comment]
> We have a few more test scenarios to add to boost coverage but not something to block this PR on.

Definitely! Targeting 90% test coverage in the next wave :)

**Pull request merged**